# Taller de bloques de código y estructuas de control

A continuación encontrarán 5 ejercicios con los cuales podrán practicar lo visto en clase. La lógica para desarrollar los mismos es libre, el todo es que cumpla con lo especificado. 

Nuevamente, la idea es que desarrollen debajo de cada literal el código correspondiente.

1. Realice un programa que reciba un número entero e imprima su descomposición en factores primos

In [0]:
b = int(input('Ingrese un número entero, por favor: '))
num_primos=[]
b1=b
cont=0
for i in range(2,b+1):
    if b1 % i == 0:
        num_primos.append(i)
#         cont=cont+1
        b1=b1/i
        
print('Los numeros primos son:')
print( num_primos)

    

Ingrese un número entero, por favor: 13195
Los numeros primos son:
[5, 7, 13, 29]


2. Realice un programa que clasifique los pesos de los **n** objetos de una bodega en tres grupos: Menos a 10 kg, entre 10 kg y 20 kg y más de 20 kg. 

  **Nota:**
  * Se debe guardar el objeto y su peso en un diccionario.
  * El usuario debe escoger qué clasificación desea. Una vez hecha la elección, el programa debe imprimir el grupo y los objetos contenidos en el mismo.

In [0]:
print('Bienvenido a bodegas Alzate')
Bodega={}
Bodega_menor_10={}
Bodega_int_1020={}
Bodega_mayor_20={}
mas='si'
while mas=='si':
  objeto=input('¿Qué dato quieres introducir?: ')
  valor=input(objeto + ':')
  Bodega[objeto] = valor
  mas = input('¿Quieres añadir más información personal (Si/No)?: ')
for objetos, peso in Bodega.items():
    peso_1=int(peso)
    if peso_1<10:
        Bodega_menor_10[objetos]=peso
    elif (peso_1>10 and peso_1<20):
        Bodega_int_1020[objetos]=peso
    elif peso_1>20:
        Bodega_mayor_20[objetos]=peso
        
print('los articulos menores de 10 Kg son: ',Bodega_menor_10)
print('los articulos entre 10 Kg y 20 Kg son: ',Bodega_int_1020)
print('los articulos mayores de 20 Kg son: ',Bodega_mayor_20)

Bienvenido a bodegas Alzate
¿Qué dato quieres introducir?: 2
2:2
¿Quieres añadir más información personal (Si/No)?: 2
los articulos menores de 10 Kg son:  {'2': '2'}
los articulos entre 10 Kg y 20 Kg son:  {}
los articulos mayores de 20 Kg son:  {}


3. Realice un programa que simule una lista de supermercado. El usuario deberá poder ingresar el artículo y su valor. Al final de ingresar los artículos, el código debera mostrar el contenido de toda la lista y el valor total a pagar. 

In [0]:
listamercado={}
mas='si'
Total=0
while mas=='si':
    
    dato=input('¿Qué producto?: ')
    print('Ingrese el valor del producto')
    valor=input(dato+':')
    
    listamercado[dato]=valor
    mas = input('¿Quieres añadir más productos?: ')
for producto,valor in listamercado.items():
    valor=int(valor)
    print('Que Cantidad del producto')
    cantidad=int(input(producto+':'))
    Total=Total+cantidad*valor
print('El total es: ',Total)
print('la lista de mercado es: ',listamercado)    
    
    
    
# print('El valor total a pagar es: ',Total)    
# print('la lista de mercado es: ',listamercado)

¿Qué producto?: silla
Ingrese el valor del producto
silla:2000
¿Quieres añadir más productos?: si
¿Qué producto?: 5000
Ingrese el valor del producto
5000:23
¿Quieres añadir más productos?: n
Que Cantidad del producto
silla:3
Que Cantidad del producto
5000:4
El total es:  6092
la lista de mercado es:  {'silla': '2000', '5000': '23'}


4. Cada nuevo término de la secuencia de Fibonacci se genera sumando los dos anteriores. Comenzando con 1 y 2, los primeros 10 términos serán:

  1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...

  Al considerar los términos de la secuencia de Fibonacci cuyos valores no superan los cuatro millones, encontrar la suma de los términos de valor par. 
  
  El resultado es: 4613732

In [0]:
# fibonacci
i=1
j=1
cont=2
Fibbo=[]
Fibbo.insert(0,i)
Fibbo.insert(1,j)
SumaPar=0
T=int(input('Ingrese el valor total que desea obtener'))
while SumaPar<T:
  Suma=i+j
  i=j
  j=Suma
  Fibbo.insert(cont,j)
  cont=cont+1
  if Suma%2==0:
    SumaPar=Suma+SumaPar
print(Fibbo)
print(SumaPar)
print('La cantidad de elementos en la secuencia de Fibonacci es: ',cont); print('Si desea compruebelo en el codigo siguiente')

Ingrese el valor total que desea obtener22
[1, 1, 2, 3, 5, 8, 13, 21, 34]
44
La cantidad de elementos en la secuencia de Fibonacci es:  9
Si desea compruebelo en el codigo siguiente


In [2]:
# fibonacci
i=1
j=1
cont=2
Fibbo=[]
Fibbo.insert(0,i)
Fibbo.insert(1,j)
SumaPar=0
T=int(input('Ingrese la cantidad de elementos de la Secuencia de Fibonacci'))
while cont<T:
  Suma=i+j
  i=j
  j=Suma
  Fibbo.insert(cont,j)
  cont=cont+1
  if Suma%2==0:
    SumaPar=Suma+SumaPar
print(Fibbo)
print('El valor de la suma de los números pares es: ',SumaPar)

Ingrese la cantidad de elementos de la Secuencia de Fibonacci5
[1, 1, 2, 3, 5]
El valor de la suma de los números pares es:  2


5. Los factores primos de 13195 son 5, 7, 13 y 29.

  ¿Cuál es el factor primo más grande del número 600851475143? 

  El resultado es: 6857

In [1]:
num=int(input('Ingrese el número del cual desee conocer sus factor primo mayor: '))
Factor_primo=int(2)

while num !=1:
    if num % Factor_primo ==0:
        num = num / Factor_primo
    else:
        Factor_primo=Factor_primo + 1

b=int(Factor_primo)
print('El factor primo más grande del numero ingresado es:',b)

Ingrese el número del cual desee conocer sus factor primo mayor: 600851475143
El factor primo más grande del numero ingresado es: 6857
